lib/readers/patter_reader_test.go

```diff
@@ -0,0 +1,30 @@
+package readers
```

Comment:
There is a typo in the filename here!  `patter_reader_test.go` should be `pattern_reader_test.go`